File: app/src/main/java/me/jerryhanks/stepviewapp/MyTimeLine.java
package me.jerryhanks.stepviewapp;

import me.jerryhanks.stepview.model.Status;
import me.jerryhanks.stepview.model.TimeLine;

/**
 * @author <@Po10cio> on 10/18/17 for StepViewApp
 */

public class MyTimeLine extends TimeLine {
    private String title;
    private String content;

    public MyTimeLine(Status status, String title, String content) {
        super(status);
        this.title = title;
        this.content = content;
    }


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof MyTimeLine)) return false;

        MyTimeLine that = (MyTimeLine) o;

        if (title != null ? !title.equals(that.title) : that.title != null) return false;
        return content != null ? content.equals(that.content) : that.content == null;
    }

    @Override
    public int hashCode() {
        int result = title != null ? title.hashCode() : 0;
        result = 31 * result + (content != null ? content.hashCode() : 0);
        return result;
    }

    @Override
    public String toString() {
        return "MyTimeLine{" +
                "title='" + title + '\'' +
                ", content='" + content + '\'' +
                '}';
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }
}

File: app/src/main/java/me/jerryhanks/stepviewapp/MyTimeLine.kt
package me.jerryhanks.stepviewapp

import me.jerryhanks.timelineview.model.Status
import me.jerryhanks.timelineview.model.TimeLine

/**
 * @author <@Po10cio> on 10/18/17 for StepViewApp
 */

 class MyTimeLine(status: Status, var title: String?, var content: String?) : TimeLine(status) {


    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as MyTimeLine

        if (title != other.title) return false
        if (content != other.content) return false

        return true
    }


    override fun hashCode(): Int {
        var result = if (title != null) title!!.hashCode() else 0
        result = 31 * result + if (content != null) content!!.hashCode() else 0
        return result
    }

    override fun toString(): String {
        return "MyTimeLine{" +
                "title='" + title + '\'' +
                ", content='" + content + '\'' +
                '}'
    }


}
